### PR TITLE
Upgrade PHP to 7.1.33 INFRA-83

### PR DESCRIPTION
**Pull request by erickhun**
No breaking changes, mostly small security updates and bug fixes https://www.php.net/ChangeLog-7.php#PHP_7_1

@colinscape up for a review? 

Also, maybe we should think a bit more on the naming the image and tags:

- For tags , I use the php minor version upgrades as an opportunity to change the image (adding libs and co)  , but it will might happen some time where there is no minor upgrade 
- For naming the image, maybe we can simply use our product names? since analyze stack != publish stack != reply stack != core stack. The downside with that is that we could endup with few base image per product 🤔 . I'm thinking something 3 base images per team : `teamX-workers` `teamX-backend` `teamX-frontend`. what will results 12 total base images for our 4 team. That's still a lot to maintain.

### CODE REVIEW

File: apache-php/7.1/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,8 @@ RUN apt-get update && \
 
 # Drivers and libraries
 RUN docker-php-ext-configure gd --with-freetype-dir=/usr/include/ --with-jpeg-dir=/usr/include/ && \
-    docker-php-ext-install -j$(nproc) gd mcrypt calendar pcntl sockets
+    docker-php-ext-configure opcache --enable-opcache && \
+    docker-php-ext-install -j$(nproc) gd mcrypt calendar pcntl sockets opcache exif
```
Comment (colinscape) [inline apache-php/7.1/Dockerfile]: Curious why we're adding opcache and exif in here as part of this. Are these needed, or are we conflating the PHP upgrade and 'sneaking' some other changes in here at the same time?

File: apache-php/7.1/Dockerfile [hunk 2]
```diff
@@ -36,6 +37,17 @@ RUN pecl install -f mongodb-1.5.3 && \
     make install && \
     docker-php-ext-enable mongodb memcached redis imagick
 
+# Configure Opcode cache https://www.scalingphpbook.com/blog/2014/02/14/best-zend-opcache-settings.html
+RUN ( \
+  echo "opcache.memory_consumption=192"; \
```
Comment (colinscape) [inline apache-php/7.1/Dockerfile]: All these settings here make we wonder if there might be unintended side-effects. Unless these are already being used in another PHP image somewhere?

**Comment by colinscape** [comment]
Regarding how we name things - that's certainly tricky. I guess it's whether we name things based on what they _are_ or on what they are _used for_. I haven't thought this through completely but I wonder if even having a named image for every service might be the way to go, likely depending on similar base images. Then we could have our very own versioning (and associated Changelog), which might make things clearer - eg bufferapp/publish-api@1.1.2. Again, this might help to 'draw the line' between what Product teams are responsible for (these named-after-service images) and what Infrastructure are responsible for (lower-level, named more like we currently have eg bufferapp/apache-php:7.1.33).

Would love to figure out a good plan for this - I can feel a project coming on! 😁 Is that something you'd be interested to take on and push forward?

File: apache-php/7.1/Dockerfile (hunk 1 above)
Comment (erickhun) [reply apache-php/7.1/Dockerfile]: mhhh yeah, maybe the `opcache` change isn't the safest thing to do here. I'll take this out for now 👍

**Comment by erickhun** [comment]
Yep, agree on naming they what they are and associating their own versioning! Sounds like a great project for 2020 @colinscape !

File: apache-php/7.1/Dockerfile (hunk 1 above)
Comment (colinscape) [reply apache-php/7.1/Dockerfile]: Awesome, thanks @erickhun ! 👍

**Comment by colinscape** [comment]
Sweet, I'll try and flesh out some of these '2020' projects over the coming weeks with my 'PM' hat on. 😀 Thanks, @erickhun ! 🌟

**Pull request merged**